src/shapes/path.class.js

```diff
@@ -237,6 +237,37 @@
       return this.path.length;
     },
 
+    /**
+     * @private
+     */
+    _projectStrokeOnSegmentBBox: function (point, pointBefore, pointAfter, bboxPoints) {
+      var v1 = pointBefore.subtract(point);
+      var v2 = pointAfter.subtract(point);
+      var angle = fabric.util.calcAngleBetweenVectors(v1, v2);
+      if (Math.abs(angle) < Math.PI / 2) {
```

Comment:
If we project all angles some bboxes of visual tests fix while other visual tests break
test:
- `clipping13` 
![image](https://user-images.githubusercontent.com/34343793/176999606-f8a3b653-04eb-4c46-b444-f1fb9fa90a6f.png)

- pathWithGradientSvg 
![image](https://user-images.githubusercontent.com/34343793/176999616-97d9c177-b644-4ed9-91c9-b8556c27eb9c.png)

- `arc3` ![image](https://user-images.githubusercontent.com/34343793/176999584-e31ee704-f2e2-4ec1-83e4-b8060555e8ed.png)